File: src/Common/src/System.Private.CoreLib/shared/System/UnhandledExceptionEventArgs.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

namespace System
{
    public class UnhandledExceptionEventArgs : EventArgs
    {
        private Object _exception;
        private bool _isTerminating;

        public UnhandledExceptionEventArgs(Object exception, bool isTerminating)
        {
            _exception = exception;
            _isTerminating = isTerminating;
        }

        public Object ExceptionObject
        {
            get { return _exception; }
        }

        public bool IsTerminating
        {
            get { return _isTerminating; }
        }
    }
}

File: src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/ApplicationServices/UnhandledExceptionEventArgs.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports System.ComponentModel
Imports System.Runtime.InteropServices
Imports System.Threading

Namespace Microsoft.VisualBasic.ApplicationServices
    ''' <summary>
    ''' Provides the exception encountered along with a flag on whether to abort the program
    ''' </summary>
    <EditorBrowsable(EditorBrowsableState.Advanced), ComVisible(False)> _
    Public Class UnhandledExceptionEventArgs : Inherits ThreadExceptionEventArgs
        Sub New(ByVal exitApplication As Boolean, ByVal exception As System.Exception)
            MyBase.new(exception)
            Me.ExitApplication = exitApplication
        End Sub

        ''' <summary>
        ''' Indicates whether the application should exit upon exiting the exception handler
        ''' </summary>
        Public Property ExitApplication() As Boolean
    End Class
End Namespace
